src/group_theory/submonoid.lean

```diff
@@ -513,138 +546,71 @@ instance has_one : has_one S := ⟨⟨_, S.one_mem⟩⟩
 
 /-- A submonoid of a monoid inherits a monoid structure. -/
 @[to_additive to_add_monoid "An `add_submonoid` of an `add_monoid` inherits an `add_monoid` structure."]
-instance to_monoid {M : Type*} [monoid M] {S : submonoid M} : monoid S :=
+instance to_monoid {M : Type*} [monoid M] (S : submonoid M) : monoid S :=
 by refine { mul := (*), one := 1, ..}; by simp [mul_assoc]
 
 /-- A submonoid of a `comm_monoid` is a `comm_monoid`. -/
 @[to_additive to_add_comm_monoid "An `add_submonoid` of an `add_comm_monoid` is an `add_comm_monoid`."]
 instance to_comm_monoid {M} [comm_monoid M] (S : submonoid M) : comm_monoid S :=
-{ mul_comm := λ _ _, subtype.ext.2 $ mul_comm _ _, ..submonoid.to_monoid}
+{ mul_comm := λ _ _, subtype.eq $ mul_comm _ _, ..submonoid.to_monoid S}
 
 /-- The natural monoid hom from a submonoid of monoid `M` to `M`. -/
 @[to_additive "The natural monoid hom from an `add_submonoid` of `add_monoid` `M` to `M`."]
-def subtype : S →* M :=
-{ to_fun := coe,
-  map_one' := rfl,
-  map_mul' := λ _ _, rfl }
-
-@[simp, to_additive] theorem subtype_apply (x : S) : S.subtype x = x := rfl
+def subtype : S →* M := ⟨coe, rfl, λ _ _, rfl⟩
```

Comment:
Wouldn't it be more natural to call this one `val`?

Reply:
I copied name from [submodule.subtype](https://github.com/leanprover-community/mathlib/blob/master/src/algebra/module.lean#L384). It's OK with me to rename it but I think that we should use the same name in both cases.

Reply:
Ok, we can change this in a later PR.